### DISCUSSION TITLE
Use python function/method within PuLP objective function

### DISCUSSION THREAD

**Discussion by Ervin66** [coin-or/pulp#406] [Q&A]
Would it be possible to define a cost function and use it while looping through the arguments within PuLP objective function?

example:

```python
def func(a, b, c):
    return a * b * c

prob += pulp.lpSum([func( x[i], y[j], z[w])
                                  for i in I
                                  for j in J
                                  for w in W])

````

I tried implementing but it seems it is not taken into consideration by the model.

**Comment by pchtsp** [comment]
Hello, I do not understand what you want to do but I recommend you follow this guide to complete your question: https://coin-or.github.io/pulp/guides/how_to_debug.html

In case x, y or z are lists or dictionaries of variables, then no: you cannot multiply variables in pulp.

**Comment by tkralphs** [comment]
I guess maybe this is basically the same  question as in #409.
